run/run.go

```diff
@@ -1,47 +1,154 @@
 package run
 
 import (
+	"encoding/base64"
+	"encoding/json"
 	"fmt"
-	"github.com/Portshift/klar/clair"
+
+	"github.com/anchore/grype/grype"
+	grype_db "github.com/anchore/grype/grype/db"
+	grype_pkg "github.com/anchore/grype/grype/pkg"
+	grype_models "github.com/anchore/grype/grype/presenter/models"
+
+	grype_client "github.com/Portshift/grype-server/api/client/client"
+	grype_client_operations "github.com/Portshift/grype-server/api/client/client/operations"
+	"github.com/Portshift/grype-server/api/client/models"
 	"github.com/Portshift/klar/config"
 	"github.com/Portshift/klar/docker"
-	log "github.com/sirupsen/logrus"
+	anchore_image "github.com/anchore/stereoscope/pkg/image"
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/format"
+	"github.com/anchore/syft/syft/source"
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
 )
 
-func ExecuteScan(conf *config.Config) ([]*clair.Vulnerability, []*docker.FsLayerCommand, error) {
-	image, err := docker.NewImage(&conf.DockerConfig)
+func ExecuteRemoteGrypeScan(imageName string, conf *config.Config) (*grype_models.Document, []*docker.FsLayerCommand, error) {
+	src, cleanup, err := source.New(imageName, &anchore_image.RegistryOptions{
+		InsecureSkipTLSVerify: conf.DockerConfig.InsecureTLS,
+		InsecureUseHTTP:       conf.DockerConfig.InsecureRegistry,
+		Credentials: []anchore_image.RegistryCredentials{
+			{
+				Authority: "", // What is this?
+				Username:  conf.DockerConfig.User,
+				Password:  conf.DockerConfig.Password,
+				Token:     conf.DockerConfig.Token,
+			},
+		},
+	})
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to parse name: %v", err)
+		return nil, nil, fmt.Errorf("failed to create syft source: %v", err)
 	}
+	defer cleanup()
 
-	err = image.Pull()
+	catalog, distro, err := syft.CatalogPackages(src, source.SquashedScope)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to catalog packages: %v", err)
+	}
+	sbomEncoded, err := syft.Encode(catalog, &src.Metadata, distro, source.SquashedScope, format.JSONOption)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to pull image: %w", err)
+		return nil, nil, fmt.Errorf("failed to encode sbom: %v", err)
 	}
 
-	if err := image.FetchFsCommands(&conf.DockerConfig); err != nil {
-		return nil, nil, fmt.Errorf("failed to fetch layer commands: %v", err)
+	sbom64 := base64.StdEncoding.EncodeToString(sbomEncoded)
+
+	client := createGrypeClient(conf.GrypeAddr)
+	params := grype_client_operations.NewPostScanSBOMParams().WithBody(&models.SBOM{
+		Sbom: sbom64,
+	})
```

Comment:
need to add `WithContext` or `WithTimeout` to support scan timeout

Reply:
I removed the scan timeout. This should be very quick and the HTTP default timeout should be enough. WDYT?

Reply:
what is the `HTTP default timeout`?

Reply:
![image](https://user-images.githubusercontent.com/39153880/144017701-dfefab6b-5d0d-4354-aeb5-9e05b26dd846.png)

Reply:
please add scan timeout and do it configurable